### DISCUSSION TITLE
TemporaryUploadedFile::dimensions(): Return value must be of type ?array, bool returned

### DISCUSSION THREAD

**Discussion by JunaidQadirB** [livewire/livewire#6426] [7. Bugs (archived)] [answered]
### Livewire version

v3.0.0-beta.10

### Which PHP version are you using?

PHP 8.1

### Repository URL

https://github.com/JunaidQadirB/livewire-3-bug

### Steps To Reproduce

1. Create a Livewire component named SubmitApp as well as its test
```bash
php artisan make:livewire SubmitApp --pest
```
2. Add a property $icon
3. Add Rule to the $icon property with image and dimensions validation
```php
    #[Rule('required|image|dimensions:width=512,height=512')]
    public $icon;

```

4. Add the following test:

```php
test('the component can validate dimensions for the icon field', function () {

    $component = Livewire::test(SubmitApp::class);

    $component
        ->set('icon', UploadedFile::fake()
            ->create('not-a-png-image.pdf', 512, 512))
        ->call('submit')
        ->assertHasErrors([
            'icon' => 'dimensions:width=512,height=512'
        ]);
});
``` 

6. Run the test
```bash
pest --filter=SubmitApp 

```
7. The test should fail to run with the following error as in the image
![image](https://github.com/livewire/livewire/assets/642353/531467f1-4531-4c9d-9651-ca9269d14153)
Text version of the error:
```
 FAILED  Tests\Feature\Livewire\SubmitAppTest > the component can…   TypeError
  Livewire\Features\SupportFileUploads\TemporaryUploadedFile::dimensions(): Return value must be of type ?array, bool returned

  at vendor/livewire/livewire/src/Features/SupportFileUploads/TemporaryUploadedFile.php:80
     76▕
     77▕     public function dimensions(): ?array
     78▕     {
     79▕         stream_copy_to_stream($this->storage->readStream($this->path), $tmpFile = tmpfile());
  ➜  80▕         return @getimagesize(stream_get_meta_data($tmpFile)['uri']);
     81▕     }
     
```

### Additional information

As we are trying to upload a file other than an image and the PHP function [`getimagesize()`](https://www.php.net/manual/en/function.getimagesize.php) returns false if the file passed to it, is not an image but the `dimensions()` function in `vendor/livewire/livewire/src/Features/SupportFileUploads/TemporaryUploadedFile.php:77` returns an optional array which throws an exception.

### Please confirm (incomplete submissions will not be addressed)

- [X] I have provided easy and step-by-step instructions to reproduce the bug.
- [X] I have provided code samples as text and NOT images.
- [X] I understand my bug report will be removed if I haven't met the criteria above.

**Comment by luanfreitasdev** [comment, marked as answer]
https://github.com/livewire/livewire/pull/6479

**Reply by JunaidQadirB** [reply]
Awesome, thanks!
